### DISCUSSION TITLE
snacks.nvim Scratch Buffer file type

### DISCUSSION THREAD

**Discussion by ibeex** [LazyVim/LazyVim#4937] [Q&A] [answered]
Hi, 
right now, when I open Scratch Buffer with `<leader>.` file type is Lua? Is there a way to change it, or it is intended behavior?

**Comment by dpetka2001** [comment]
You can pass `ft` to `Snacks.scratch`. Check the docs of `snacks.nvim` and also search discussions/issues on the repo. It has been asked before but i can't remember off the top of my head. But you can definitely do what you want.

**Comment by dpetka2001** [comment]
`require("snacks").scratch({ ft = "your_filetype" })` should do what you want i think.

**Comment by folke** [comment, marked as answer]
It's indeed not ideal. I'm thinking of changing it roughly to something like `Snacks.scratch({ft = vim.bo.filetype})`

**Reply by ibeex** [reply]
That would be just what I was thinking when I have seen this feature. Thanks!

**Reply by folke** [reply]
Yeah, it just gets tricky if you open a scratch buffer from the dashboard or something.
In that case, a scratch buffer with the `snacks_dashboard` filetype doesn't really make sense.
I can easily check for `buftype`, but then what ft do I open?

**Reply by dpetka2001** [reply]
Why don't you just leave the `ft` not set when `buftype = nofile`? Doesn't make sense to set a filetype in that case does it? User can just set the filetype himself if he wants afterwards. But maybe check with an autocmd on close if the filetype still is not set? Because if the user set it himself in the meantime, then that should persist in the next invocation (except if it already does this, I haven't tested).

**Reply by folke** [reply]
I was thinking of doing it like this:
![image](https://github.com/user-attachments/assets/bc3ec9a0-7f66-4b0a-be36-52328358d473)

This would get multiple scratch buffers for one project based on what file you open it in.

Wdyt?

**Reply by dpetka2001** [reply]
So either `vim.bo.filetype` or markdown as fallback. Sounds sane default to me. I'm guessing one can overwrite that `ft` function for a fallback of his own right?

**Comment by rjshrjndrn** [comment]
For any future wanderers, something like this will do
```lua
return {
  "folke/snacks.nvim",
  opts = {
    dashboard = { enabled = false },
    scratch = {
      ft = "markdown",
      cmd = "Scratch",
      name = "Scratch",
    },
  },
}
```
